src/components/AddBook.js

```diff
@@ -1,14 +1,55 @@
-import React from 'react';
-
-const AddBook = () => (
-  <section className="add-book-form">
-    <h2>ADD NEW BOOK</h2>
-    <form className="books-list">
-      <input type="text" placeholder="Author" />
-      <input type="text" placeholder="Book title" />
-      <button type="button">ADD BOOK</button>
-    </form>
-  </section>
-);
+import { useDispatch, useSelector } from 'react-redux';
+import React, { useState } from 'react';
+import { addBookAction } from '../redux/books/books';
+
+const AddBook = () => {
+  const books = useSelector((state) => state.books);
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
+  const [author, setAuthor] = useState('');
+
+  const handleAuthorChange = (e) => {
+    setAuthor(e.target.value);
+  };
+
+  const handleTitleChange = (e) => {
+    setTitle(e.target.value);
+  };
+
+  const returnState = (e) => {
+    e.preventDefault();
+    const obj = {
+      id: books.length,
+      author,
+      title,
+    };
+    dispatch(addBookAction(obj));
+    e.target.reset();
+  };
+
+  return (
+    <section className="add-book-form">
+      12,n
```

Comment:
- [ ] **[OPTIONAL]:** I think that you misplaced this piece of code here, if that's the case, please delete it since it can cause some readability issues! 🤓 🛠️